File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/constant/Constant1.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Bool import Bool

@PythonTaskInfo(constant=True)
class Constant1(PythonTask):

    def __init_ports__(self):
        self.out = OutputPort(name="Out", data_type=Bool)

    def execute(self, grp):
        self.out.put_data(Bool([False, True], [1, 2]))

File: com.systemincloud.examples.tasks.rtask/src/test/r/constant/Constant1.R

library(R6)
library(sicrtask)

Constant1 <- R6Class("Constant1",
  inherit = RTask,
  public = list(
    rtaskinfo = RTaskInfo$new(constant = TRUE),
    out = NA,
    initialize_ports = function() {
      self$out <- OutputPort$new(name = "out", data_type = "Bool")
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  self$out$put_data(Bool$new(c(FALSE, TRUE), c(1, 2)))
    }
  ),
  private = list(
  )
)
